spec/models/stats/stats_report_spec.rb

```diff
@@ -123,47 +123,6 @@
           .to change { described_class.completed.where(report_name: 'my_new_report').first }
           .to report
       end
-
-      it 'sets the checksum' do
-        expect { write_report }.to change(report, :as_document_checksum).to(checksum)
-      end
-    end
-  end
-
-  describe '#document_url' do
-    context 'when document is nil' do
-      let(:report) { build(:stats_report, report: nil, document: nil) }
-
-      it { expect(report.document_url).to be_nil }
-    end
-
-    context 'when document exists' do
-      let(:report) { build(:stats_report, :with_document) }
-
-      context 'when the document storage is filesystem' do
-        let(:options) { { storage: :filesystem } }
-
-        before do
-          allow(report.document).to receive(:options).and_return(options)
-        end
-
-        it 'returns the document path' do
-          expect(report.document_url).to eq('tmp/test/reports/report.csv')
-        end
-      end
-
-      context 'when the document storage is S3' do
-        let(:options) { { storage: :s3 } }
-
-        before do
-          original_options = report.document.options
-          allow(report.document).to receive(:options).and_return(original_options.merge(options))
-        end
-
-        it 'returns the an expiring url for the document' do
```

Comment:
Is there anyway to emulate this test for an expired URL? - I assume, perhaps wrongly, that it is

Reply:
This is what I was attempting at line 41 of spec/controllers/case_workers/admin/management_information_controller_spec.rb

Reply:
> as discussed, testing AS internals is probably overkill